code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/AbstractTokenUtil.java

```diff
@@ -222,10 +222,9 @@ public boolean findAndSetJWT() {
 
     @Override
     public boolean isAllowed(List<String> idList, Set<Identity> identities) {
-        boolean hasAccessToAProject = authDao.hasAccessToAnyProject(identities, false, null);
         switch (accessMode()) {
             case "restricted":
-                if (hasAccessToAProject || isWhitelisted(idList)) {
+                if (isWhitelisted(idList)) {
```

Comment:
isWhitelisted() implies that the user can login to rancher. Originally even if the user was not in the whitelist, he was alllowed to login if he was invited to some project.
Now we want to let the user login only if he belongs to the whitelist, when "restricted" access is set.

Problem with using hasAccessToAProject && isWhitelisted(idList), would when the user is attempting to login for the very first time, his personal project is not created at this point I guess. So there many not be any project he has access to.